lib/streamlit/dataframe_util.py

```diff
@@ -87,13 +89,43 @@
 _DASK_INDEX: Final = "dask.dataframe.core.Index"
 _RAY_MATERIALIZED_DATASET: Final = "ray.data.dataset.MaterializedDataset"
 _RAY_DATASET: Final = "ray.data.dataset.Dataset"
+_DUCKDB_RELATION: Final = "duckdb.duckdb.DuckDBPyRelation"
 
 V_co = TypeVar(
     "V_co",
     covariant=True,  # https://peps.python.org/pep-0484/#covariance-and-contravariance
 )
 
 
+@runtime_checkable
+class DBAPICursor(Protocol):
+    """Protocol for DBAPI 2.0 Cursor objects (PEP 249).
+    This is a simplified version of the DBAPI Cursor protocol.
```

Comment:
nit: could you leave a sentence please why the simplified version is enough for us? I assume we implement only the ones required for reading data within our commands

Reply:
Yep, only the methods are relevant for our integration. I added that to the comment.